core/src/main/java/io/micronaut/core/annotation/AnnotationValueBuilder.java

```diff
@@ -103,6 +107,9 @@ public AnnotationValue<T> build() {
     @NonNull
     public AnnotationValueBuilder<T> stereotype(AnnotationValue<?> annotation) {
         if (annotation != null) {
+            if (stereotypes == null) {
+                stereotypes = new ArrayList<>();
```

Comment:
Perhaps limit the array size since there are generally no more than say 10 stereotypes